controllers/goflowkube/goflowkube_objects.go

```diff
@@ -72,18 +71,33 @@ func buildDaemonSet(desired *flowsv1alpha1.FlowCollectorGoflowKube, ns string) *
 			Selector: &metav1.LabelSelector{
 				MatchLabels: buildLabels(),
 			},
-			Template: *buildPodTemplate(desired),
+			Template: buildPodTemplate(desired, true),
 		},
 	}
 }
 
-func buildPodTemplate(desired *flowsv1alpha1.FlowCollectorGoflowKube) *corev1.PodTemplateSpec {
+func buildPodTemplate(desired *flowsv1alpha1.FlowCollectorGoflowKube, forDaemonSet bool) corev1.PodTemplateSpec {
 	cmd := buildMainCommand(desired)
-	return &corev1.PodTemplateSpec{
+	var ports []corev1.ContainerPort
+	var tolerations []corev1.Toleration
+	if forDaemonSet {
```

Comment:
can't we use `desired.Kind == constants.DaemonSetKind` here instead of an additionnal parameter ?

Reply:
Sure. Changing